pkg/manager/volumes/pvc_replacer.go

```diff
@@ -73,6 +73,11 @@ func (p *pvcReplacer) getVolReplaceStatusForComponent(tc *v1alpha1.TidbCluster,
 }
 
 func (p *pvcReplacer) UpdateStatus(tc *v1alpha1.TidbCluster) error {
+	if !tc.IsPVCReplaceEnabled() {
```

Comment:
Instead of disabling here, can we do it at the same place as the existing feature flag, just so both pieces of code disabling this are next to each other?

The gate for that is here: https://github.com/pingcap/tidb-operator/blob/9ef26f88bf75338cef427b44aa6303237ddcdbd6/pkg/controller/tidbcluster/tidb_cluster_control.go#L183C41-L183C65
and here: 
https://github.com/pingcap/tidb-operator/blob/9ef26f88bf75338cef427b44aa6303237ddcdbd6/pkg/controller/tidbcluster/tidb_cluster_control.go#L309

Reply:
I am okay to move the check however I kept it inside considering it declutters the tidb_cluster_control.go file and it follows a similar approach as `enbalePVReclaim`. Feature gate probably makes sense at tidb_cluster_control.go level as if the feature itself is disabled why should we even go to pvc_cleaner?
https://github.com/pingcap/tidb-operator/blob/9ef26f88bf75338cef427b44aa6303237ddcdbd6/pkg/manager/member/pvc_cleaner.go#L75

Let me know if you want me to still move it to tidb_cluster_control.go?

Reply:
personally I feel easier to read the code if both disables are in the same place since they functionally do the same thing. but no strong preference eitherways. I'll leave it to @csuzhangxc

Reply:
@csuzhangxc please check and let me know if this is fine or do you want me to move the check at tidb_cluster_control.go?

Reply:
I prefer to move the check to tidb_cluster_control.go.

BTW, can we define the behavior as:
- if one of these feature-gates is enabled, this feature is enabled
- only if both of them are disabled, this feature is disabled

Reply:
Ok I will move the check

However 2nd part doesn't make sense when we want to enable TiDB operator to handle volume replace feature but allow only if the TiDB cluster has this feature flag enabled.

Consider a scenario where 1 tidb-operator is managing 5 tidb clusters (1, 2, 3, 4, 5). We want to enable volume replace only for cluster 3. If we enable the feature gate it would enable it for all the clusters. 

Also can you let me know the reason why this has not been moved to GA? can you share the issues which needs to be fixed? 

One such issue I know is handling of TiFlash is not enabled which I will raise PR in next week.

Reply:
I think what @csuzhangxc is suggesting would give a path to eventually deprecate the feature-gate and only rely on this new tc-level setting. I'd be fine with dropping the feature-gate entirely in favor of this new tc-level setting alone if that's an option but may cause compatibility issues for anyone else relying on flag, will let Xuecheng decide.

TiCDC is one other component I did not get a chance to add support for VolumeReplace in my original PR

Reply:
@anish-db I am aligned on the removal of the feature gate once it's GA. I would love to add support for both TiFlash and TiCDC in next PR. I hope that's fine.

Reply:
@anish-db @csuzhangxc After giving a second thought, the suggestion from @csuzhangxc makes sense.

It makes this new feature gate backward compatible. And in our scenario where we want to enable it for specific clusters we can keep the feature gate disabled and only enable tc level flag. 

I have made the changes please review again